tests/phpunit/tests/test-instrumentation.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+require_once __DIR__ . '/../mockfactory/Request.php';
+require_once __DIR__ . '/../../../includes/plugin-instrumentation.php';
+
+use Razorpay\MockApi\MockApi;
+
+class Test_Instrumentation extends WP_UnitTestCase
+{
+    private $rzpInstrumentation;
+
+    public function setup(): void
+    {
+        parent::setup();
+        $this->rzpInstrumentation = Mockery::mock('TrackPluginInstrumentation',
+            ['key_id', 'key_secret'])->makePartial()->shouldAllowMockingProtectedMethods();
+    }
+
+    public function testHooks()
+    {
+        $this->assertSame(10, has_action('activated_plugin', 'razorpayPluginActivated'));
+        $this->assertSame(10, has_action('deactivated_plugin', 'razorpayPluginDeactivated'));
```

Comment:
This test case is failing for the hooks i have changed in last release
[https://github.com/razorpay/razorpay-woocommerce/blob/master/woo-razorpay.php#L37-L39
](https://github.com/razorpay/razorpay-woocommerce/blob/master/woo-razorpay.php#L37-L39
)

Reply:
@sonyy-m register_activation_hook and register_deactivation_hook will differ the hook name like activate_plugin_dir/plugin.php will change the hook name